### PR TITLE
Mark honest/malicious pointers during honest decision

### PR DESCRIPTION
**Pull request by zjmiller**

### CODE REVIEW

**Comment by zjmiller** [comment]
image

**Comment by zjmiller** [comment]
image

**Comment by zjmiller** [comment]
image

File: client/src/lib/slate-pointers/PointerImportNode.tsx [hunk 1]
```diff
@@ -282,7 +317,11 @@ class PointerImportNodePresentational extends React.Component<any, any> {
         transition: "background-color 0.4s",
         whiteSpace: "nowrap",
         ":hover": {
-          backgroundColor: isLazyPointer
+          backgroundColor: isKnownAsMaliciousAnswerCandidate
+            ? "red"
+            : isKnownAsHonestAnswerCandidate
+            ? "green"
+            : isLazyPointer
             ? "red"
```
Comment (andrewschreiber) [inline client/src/lib/slate-pointers/PointerImportNode.tsx]: Noticing the same `red` is being used for `isKnownAsMaliciousAnswerCandidate` and `isLazyPointer` here (and above). I think its good practice to use different colors for each state if possible.

@brachbach do you have any color preferences here?

File: client/src/lib/slate-pointers/PointerImportNode.tsx [hunk 2]
```diff
@@ -213,8 +231,18 @@ class PointerImportNodePresentational extends React.Component<any, any> {
...
+      isAwaitingHonestExpertDecision,
     } = this.props;
 
+    const exportPointerId = this.props.nodeAsJson.data.pointerId;
+    const isKnownAsHonestAnswerCandidate =
+      isAwaitingHonestExpertDecision &&
+      idOfHonestAnswerCandidate === exportPointerId;
+    const isKnownAsMaliciousAnswerCandidate =
+      isAwaitingHonestExpertDecision &&
+      idOfMaliciousAnswerCandidate === exportPointerId;
```
Comment (andrewschreiber) [inline client/src/lib/slate-pointers/PointerImportNode.tsx]: nitpick: swap the order of 240 & 243, so it matches the logic chain order below.

File: client/src/lib/slate-pointers/PointerImportNode.tsx [hunk 3]
```diff
@@ -258,7 +285,11 @@ class PointerImportNodePresentational extends React.Component<any, any> {
     const styles = StyleSheet.create({
       OuterPointerImportStyle: {
         ":before": {
-          backgroundColor: isLazyPointer
+          backgroundColor: isKnownAsMaliciousAnswerCandidate
+            ? "red"
+            : isKnownAsHonestAnswerCandidate
+            ? "green"
+            : isLazyPointer
```
Comment (andrewschreiber) [inline client/src/lib/slate-pointers/PointerImportNode.tsx]: I see use of both `isLazyPointer` and `isLazy` for this same logic. Do you think we could use the same name or is there a good reason for them to be named differently?

File: server/lib/models/workspace.ts [hunk 4]
```diff
@@ -189,6 +189,95 @@ export default class Workspace extends Model<Workspace> {
...
+
+      const idOfA1AnswerCandidate = _.get(
+        judgeQuestionBlock,
+        "value[0].nodes[3].data.pointerId",
+      );
+
+      const idOfA2AnswerCandidate = _.get(
+        judgeQuestionBlock,
+        "value[0].nodes[5].data.pointerId",
+      );
```
Comment (andrewschreiber) [inline server/lib/models/workspace.ts]: Need some checks for these. Your approach in the main PR works great

File: server/lib/models/workspace.ts [hunk 5]
```diff
@@ -189,6 +189,95 @@ export default class Workspace extends Model<Workspace> {
...
+      if (isExpertWorkspace) {
+        return;
+      }
+
+      // The pointer containing the malicious answer candidate
+      // first appears in the judge question. But to distinguish
+      // it from the honest answer candidate, we need to know which pointer
+      // contains the honest answer candidate, and this first appears in the
+      // malicious question.
+      const maliciousWorkspace = await Workspace.findByPk(workspace.parentId);
```
Comment (andrewschreiber) [inline server/lib/models/workspace.ts]: Great comment, thanks!

File: client/src/lib/slate-pointers/PointerImportNode.tsx (hunk 1 above)
Comment (zjmiller) [reply client/src/lib/slate-pointers/PointerImportNode.tsx]: We don't really use lazy pointers at the moment, and the context would presumably disambiguate for the user. But curious what @brachbach thinks.

**Pull request merged**